### DISCUSSION TITLE
Gamma.InvCDF

### DISCUSSION THREAD

**Discussion by bojtib** [mathnet/mathnet-numerics#825] [Q&A] [answered]
Returns other value than excel and OpenOffice.
E.g.:

- probability: 0.12
- alfa: 2
- beta: 3

**OpenOffice and Excel result:** 1.78264358
**Mathnet result:** 0.19807150889941841

MathNet.Numerics version: 3.17.0

[http://accord-framework.net/docs/html/M_Accord_Statistics_Distributions_Univariate_GammaDistribution_InverseDistributionFunction.htm](url) also gives  1.78264358

**Comment by cdrnet** [comment, marked as answer]
The Gamma distribution can be parameterized either by a shape (alpha) and a rate (beta) or a shape (alpha) and a scale (theta), where the scale is the inverse of the rate. Math.NET offers both when using an instance of the distribution, but the static members offer only the default one, which is shape and rate.

Given your numbers it seems to me what you refer to as `beta` is actually a scale, not a rate as hinted by the name. You can either convert your scale to a rate by inverting it, or use an instance to make it explicit:

```
Gamma.InvCDF(2.0, 1/3.0, 0.12) // static function
Gamma.WithShapeScale(2.0, 3.0).InverseCumulativeDistribution(0.12) // instance method
```

Both return `1.78264358009477`.
